pkg/channel/distributed/receiver/channel/channel.go

```diff
@@ -89,7 +89,7 @@ func ValidateKafkaChannel(channelReference eventingChannel.ChannelReference) err
 	}
 
 	// Check KafkaChannel READY Status
-	if !kafkaChannel.Status.IsReady() {
+	if !kafkaChannel.IsReady() {
```

Comment:
I think this needs to be back to `!kafkaChannel.Status.IsReady()` so that it is NOT including the generation.  I don't want the receiver to stop producing messages if the channel is being reconciled.

Reply:
Travis, can you please explain the reasoning for this decision?

IIUC, if I make a mistake in the configuration like the bootstrap server is wrong, all channels will become _not_ ready (because that's a global configuration), so my data plane stops producing messages for all channels.

Given the above consideration, should we prevent taking down the data plane components and keep using a previous valid configuration while providing meaningful error messages to users at the control plane level to fix the issue?

Reply:
Hi Pier,

Sure, happy to describe my concern - it might be invalid as I don't fully understand all the potential lifecycle possibilities related to how/when the Observed Generation might become out of sync.  Thanks for bearing with me and educating me!

**First some background on the "distributed" Kafka Channel implementation...**

The single Kafka Receiver Deployment is separate from the Dispatcher Deployments for which there are one per KafkaChannel.  The Receiver receives all events and writes them to the appropriate Kafka Topics.  The line referenced above is in the `ValidateKafkaChannel()` function which checks the KafkaChannel Status before attempting to send the event to the Kafka Topic.

**The scenario I was worried about is...**

If we have a functioning KafkaChannel that is actively processing events, I wouldn't want it to start rejecting events simply because of a temporary generation mismatch in the KafkaChannel due the Controller having to cycle again to update it.

The way I was thinking (maybe totally wrong?) the generation would be changed by K8S and then our controller would update the Observed Generation and so I cannot have events fail in that interim period.

If we are certain that the normal updating of KafkaChannels and their status won't result in these temporary inconsistencies then I guess I am ok with the change.  I haven't pulled down this branch to experiment with various lifecycle change to convince myself - and clearly I'm not as knowledgable as you or Matthias on the matter ; ).

Reply:
I agree with you that we should not consider generation numbers here, but I do think we shouldn't check for readiness here at all.

Imagine this scenario:
- all channels are ready and serving traffic
- I mistakenly change bootstrap server to be non a valid Kafka server
- controller reconciles channels and all of them become not ready
- this checks here becomes true, so every channel stops serving traffic

I might missing something about how do you observe channel updates but this seems a check that should be in a different place.

Reply:
Yeah, it's interesting you bring that up ; ).  There are a couple of related issues around the general Status tracking of distributed KafkaChannels...

1. When we converted the distributed KafkaChannel to use the same CRD as the consolidated implementation we inherently lost some fidelity in the tracking of the Status due to our runtime deployment architecture differences. 
2. The consolidated channel added the "prober" status tracking which we don't have in the distributed implementation, so we're not tracking each Dispatcher Replica correctly. 

In the scenario you described above, the Controller fails to create the Kafka AdminClient (because the brokers are wrong) and thus can't re-reconcile the KafkaChannels and their Status does not change.  Both KafkaChannel implementation re-create the AdminClient every reconciliation loop in order to work around Sarama issues with long-lived connections.

Given that this distributed Receiver implementation has been in-place for long time, and is working as designed, I would vote that this PR not modify the current Receiver logic.  It's on my personal list of Issues to try and address the above inconsistencies, but it's not the highest priority.  I might not be against removing this check but it shouldn't be tied up with this PR (imo).  Let me know what you think - thanks again!